launchable/test_runners/gradle.py

```diff
@@ -104,4 +106,22 @@ def to_class_file(class_name: str):
     return class_name.replace('.', '/') + '.class'
 
 
-record_tests = launchable.CommonRecordTestImpls(__name__).report_files()
+@click.argument('reports', required=True, nargs=-1)
+@launchable.record.tests
+def record_tests(client, reports):
+    default_path_builder = client.path_builder
+
+    def path_builder(case: TestCase, suite: TestSuite,
+                     report_file: str) -> TestPath:
+        """
+        With @Nested tests in JUnit 5, test class names have inner class names
+        like com.launchableinc.rocket_car_gradle.AppTest$InnerClass.
+        It causes a problem in subsetting bacause Launchable CLI can't detect inner classes in subsetting.
+        So, we need to ignore the inner class names. The inner class name is separated by $.
+        """
+        test_path = default_path_builder(case, suite, report_file)
+        return [{**item, "name": item["name"].split("$")[0]} for item in test_path]
```

Comment:
We can use the `$` to the class name. I know it's a rare case so you don't have to consider it.
But could you leave a comment about the cli doesn't care it, for now

Reply:
https://github.com/launchableinc/cli/pull/862/commits/604fce7fee6a435d4045e2ad32f7aab790526ae0